core/src/main/java/nl/nn/adapterframework/jndi/PoolingJndiDataSourceFactory.java

```diff
@@ -65,13 +66,23 @@ public PoolingJndiDataSourceFactory() {
 	protected DataSource augmentDatasource(CommonDataSource dataSource, String dataSourceName) {
 		if(maxPoolSize > 1) {
 			log.info("Creating connection pool for datasource [{}]", dataSourceName);
-			return createPool((DataSource)dataSource);
+			return createPool((DataSource)dataSource, dataSourceName);
 		}
 		log.info("Pooling not configured, using datasource [{}] without augmentation", dataSourceName);
 		return (DataSource) dataSource;
 	}
 
-	protected DataSource createPool(DataSource dataSource) {
+	private static boolean isPooledDataSource(CommonDataSource dataSource) {
+		return dataSource instanceof ConnectionPoolDataSource
+				|| dataSource.getClass().getName().startsWith("org.apache.tomcat")
+				;
```

Comment:
At least we fix the Tomcat issues with this solution, which is the most important application server used by F!F in the wild. 

Cosmetic minor remark: this can also be put at one line?

Also something like this can be added: 
`// Datasource from Tomcat is already doing pooling` or perhaps that's already obvious.
Or a Javadoc before the method: `True if Tomcat connection pooling is used`